src/main/java/com/cognifide/aemrules/checks/slingquery/SlingQueryImplicitStrategyCheck.java

```diff
@@ -0,0 +1,175 @@
+import com.cognifide.aemrules.version.AemVersion;
+import java.util.HashMap;
+import java.util.Map;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ExpressionStatementTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+@Rule(
+    key = SlingQueryImplicitStrategyCheck.RULE_KEY,
+    name = SlingQueryImplicitStrategyCheck.RULE_MESSAGE,
+    priority = Priority.MINOR,
+    tags = Tags.AEM
+)
+@AemVersion(
+    all = true
+)
+public class SlingQueryImplicitStrategyCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+    protected static final String RULE_KEY = "AEM-19";
+
+    protected static final String RULE_MESSAGE = "Implicit search strategy used in Sling Query";
+
+    private static final String FIND_METHOD = "find";
+
+    private static final String SEARCH_STRATEGY_METHOD = "searchStrategy";
+
+    private static final String APACHE_SLING_QUERY = "org.apache.sling.query.SlingQuery";
+
+    private static final String SLING_QUERY = "SlingQuery";
+
+    private static final String DOLLAR_SIGN = "$";
+
+    private String currentSlingQueryNameVariable = null;
+
+    private Map<String, SlingQueryStates> slingQueries = new HashMap<>();
+
+    private boolean findMethodUsed = false;
+
+    private boolean searchStrategyMethodUsed = false;
+
+    private boolean ignoreIssues = false;
+
+    private JavaFileScannerContext context;
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+        this.context = context;
+        scan(context.getTree());
+    }
+
+    @Override
+    public void visitVariable(VariableTree tree) {
+        if (isSlingQuery(tree)) {
+            slingQueries.put(tree.simpleName().name(), SlingQueryStates.NOT_USED);
+            currentSlingQueryNameVariable = tree.simpleName().name();
+        }
+        super.visitVariable(tree);
+        // This part of code will be executed directly after initialization
+        if (findWithoutStrategyWasUsedOnSlingQuery()) {
+            context.reportIssue(this, tree, RULE_MESSAGE);
+            slingQueries.put(currentSlingQueryNameVariable, SlingQueryStates.ISSUE_RETURNED);
+            ignoreIssues = true;
+        }
+    }
+
+    @Override
+    public void visitExpressionStatement(ExpressionStatementTree tree) {
+        currentSlingQueryNameVariable = tree.firstToken().text();
+        if (isThisANewSlingQuery()) {
+            slingQueries.putIfAbsent(currentSlingQueryNameVariable, SlingQueryStates.NOT_USED);
+        }
+        super.visitExpressionStatement(tree);
+        // This part of code will be executed directly after expression
+        if (findWithoutStrategyWasUsedOnSlingQuery()) {
```

Comment:
There is a code duplication in here. (the whole if). You can created method like `reportIssueIfStrategyNotUsed(tree)`